File: www/cgi-bin/echo.php
#!/usr/bin/env php
<?php
// CGI response headers
echo "Content-Type: text/html\r\n\r\n";

echo "<html><body>\n";
echo "<h1>CGI Demo</h1>\n";

// Handle GET parameters
if ($_SERVER['REQUEST_METHOD'] === 'GET') {
    echo "<p><b>GET parameters:</b> Hello from GET body</p>\n";
}

// Handle POST body
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    // Uncomment these lines if you want to actually read POST data:
    // $post_data = file_get_contents("php://stdin");
    // echo "<p><b>POST body:</b> $post_data</p>";
    echo "<p><b>POST body:</b> Hello from POST body</p>\n";
}
echo "</body></html>\n";
echo "</body></html>";
?>

File: www/cgi-bin/echo.py
#!/usr/bin/env python3
import os
import sys

# CGI response headers
print("Content-Type: text/html\r\n\r\n")

print("<html><body>")
print("<h1>CGI Demo</h1>")

# Handle GET parameters (from QUERY_STRING)
if os.environ.get("REQUEST_METHOD", "") == "GET":
    print(f"<p><b>GET parameters:</b> Hello from GET body</p>")

# Handle POST body (from stdin)
if os.environ.get("REQUEST_METHOD", "") == "POST":
#    content_length = int(os.environ.get("CONTENT_LENGTH", 0))
#    post_data = sys.stdin.read(content_length)
    print(f"<p><b>POST body:</b> Hello from POST body</p>")

print("</body></html>")
